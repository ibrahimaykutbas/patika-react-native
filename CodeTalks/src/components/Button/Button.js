import React from 'react';
import {TouchableOpacity, Text, ActivityIndicator} from 'react-native';

import styles from './Button.style';

const Button = ({title, theme = 'primary', onPress, loading}) => {
  return (
    <TouchableOpacity style={styles[theme].container} onPress={onPress}>
      <Text style={styles[theme].text}>
        {loading ? <ActivityIndicator size="small" color="white" /> : title}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
