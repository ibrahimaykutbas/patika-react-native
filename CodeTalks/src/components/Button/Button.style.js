import {StyleSheet} from 'react-native';

const baseStyle = StyleSheet.create({
  container: {
    width: '100%',
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    marginTop: 20,
  },
  text: {
    fontWeight: 'bold',
  },
});

export default {
  primary: StyleSheet.create({
    container: {
      ...baseStyle.container,
      backgroundColor: '#ffa270',
    },
    text: {
      ...baseStyle.text,
      color: 'white',
    },
  }),

  secondary: StyleSheet.create({
    container: {
      ...baseStyle.container,
      backgroundColor: 'white',
    },
    text: {
      ...baseStyle.text,
      color: '#ff7043',
    },
  }),
};
