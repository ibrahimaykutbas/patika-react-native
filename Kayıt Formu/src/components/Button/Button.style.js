import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: '#664E88',
    borderRadius: 5,
    padding: 10,
    marginHorizontal: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
    fontWeight: 'bold',
  },
});
