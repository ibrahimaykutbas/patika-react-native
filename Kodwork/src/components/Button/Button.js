import React from 'react';
import {TouchableOpacity, Text} from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

import styles from './Button.style';

const Button = ({title, iconName}) => {
  return (
    <TouchableOpacity style={styles.container}>
      <Icon name={iconName} size={25} color="white" style={styles.icon} />
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;
