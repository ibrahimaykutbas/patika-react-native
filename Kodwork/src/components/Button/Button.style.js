import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: '#f44336',
    minWidth: 175,
    height: 40,
    borderRadius: 5,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 5,
    marginHorizontal: 5,
  },
  text: {
    color: 'white',
    alignSelf: 'center',
    textAlign: 'center',
  },
  icon: {
    alignSelf: 'center',
    alignItems: 'center',
    paddingRight: 5,
  },
});
