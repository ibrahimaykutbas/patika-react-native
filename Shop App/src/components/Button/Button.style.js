import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    padding: 10,
    marginHorizontal: 10,
    marginVertical: 5,
    borderRadius: 5,
    alignItems: 'center',
    backgroundColor: '#2286c3',
  },
  title: {
    fontWeight: 'bold',
    fontSize: 17,
    color: 'white',
  },
});
