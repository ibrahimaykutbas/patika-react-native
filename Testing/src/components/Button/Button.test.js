import React from 'react';

import {render, fireEvent} from '@testing-library/react-native';

import Button from './Button';

import styles from './Button.style';

test('should match with snapshot', () => {
  const comp = render(<Button />);
  expect(comp).toMatchSnapshot();
});

test('should render title correctly', () => {
  const testTitle = 'Test Title';
  const comp = render(<Button title={testTitle} />);

  const buttonText = comp.getByTestId('button-title').children[0];
  expect(buttonText).toBe(testTitle);
});

test('should trigerr onPress', () => {
  const mockFunction = jest.fn();
  const comp = render(<Button onPress={mockFunction} />);

  const buttonTouchable = comp.getByTestId('button-touchable');
  fireEvent(buttonTouchable, 'press');

  expect(mockFunction).toBeCalled();

  /* 
    Eğer fonksiyonun belli bir sayıda tetiklenmesini istiyorsak:
    expect(mockFunction).toBeCalledTimes(1);

    Not:
    fireEvent'i de test sayısı kadar oluşturmamız gerekiyor.
  */
});

test('should render default theme style', () => {
  const comp = render(<Button />);

  const buttonStyle = comp.getByTestId('button-touchable').props.style;

  expect(buttonStyle).toMatchObject(styles.primary.container);
});

test('should render given theme style', () => {
  const selectedTheme = 'secondary';

  const comp = render(<Button theme={selectedTheme} />);

  const buttonStyle = comp.getByTestId('button-touchable').props.style;

  expect(buttonStyle).toMatchObject(styles[selectedTheme].container);
});
