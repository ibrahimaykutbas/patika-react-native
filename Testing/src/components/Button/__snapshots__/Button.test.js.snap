// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 10,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    testID="button-title"
  />
</View>
`;
